FlatCache/FlatCache.swift

```diff
@@ -128,13 +129,33 @@ public final class FlatCache {
             }
         }
     }
-
+    
     public func get<T: Cachable>(id: String) -> T? {
         assert(Thread.isMainThread)
-
+        
         let key = FlatCacheKey(typeName: T.typeName, id: id)
         return storage[key] as? T
     }
-
+    
+    /// Removes an item from cache.
+    /// - Important: This method must be called from the main thread.
+    /// - Parameter id: The key corresponding to the cached item.
+    /// - Returns: An optional `Cachable` object if it exists.
+    /// - Throws: Throws a `FlatCacheError` if no value could be found for the given key.
+    public func remove<T: Cachable>(key id: String) throws -> T? {
+        assert(Thread.isMainThread)
+        
+        let key = FlatCacheKey(typeName: T.typeName, id: id)
+        guard let val = storage[key] as? Cachable else {
+            throw FlatCacheError.noValueForKey(id)
+        }
+        storage.removeValue(forKey: key)
+        enumerateListeners(key: key) { listener in
+            listener.flatCacheDidUpdate(cache: self, update: FlatCache.Update.removeItem(val))
+            if let index = listeners.index(forKey: key) {
+                listeners.remove(at: index)
+            }
+        }
+        return val as? T
+    }
```

Comment:
What information does the `throw` provide that a `nil` return value doesn't?

Reply:
So the information that the error would throw that would be useful is the key that there was no value for.  It then becomes a question of is it useful to do the try catch dance as opposed to just outright passing up the optional value.  What do you think?